packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -375,6 +375,13 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
         _validateRelayDataHash(depositHash, relayData);
         require(relayData.relayState == RelayState.Pending, "Already settled");
 
+        // Note: this check is to give the relayer a small, but reasonable amount of time to complete the relay before
+        // before it can be "stolen" by someone else. This is to ensure there is an incentive to settle relays quickly.
+        require(
+            msg.sender == relayData.slowRelayer || getCurrentTime() > request.expirationTime + 15 minutes,
```

Comment:
Note: one good thing about this check (gas-wise) is that we only pay for the second part of the check if the the original relayer is not the caller (rare).